packages/artifact/src/internal/download-http-client.ts

```diff
@@ -81,22 +85,44 @@ export class DownloadHttpClient {
   async downloadSingleArtifact(downloadItems: DownloadItem[]): Promise<void> {
     const DOWNLOAD_CONCURRENCY = getDownloadFileConcurrency()
     // limit the number of files downloaded at a single time
+    debug(`Download file concurrency is set to ${DOWNLOAD_CONCURRENCY}`)
     const parallelDownloads = [...new Array(DOWNLOAD_CONCURRENCY).keys()]
+    let currentFile = 0
     let downloadedFiles = 0
+
+    info(
+      `Total number of files that will be downloaded: ${downloadItems.length}`
+    )
+
+    this.statusReporter.setTotalNumberOfFilesToProcess(downloadItems.length)
+    this.statusReporter.start()
+
     await Promise.all(
       parallelDownloads.map(async index => {
-        while (downloadedFiles < downloadItems.length) {
-          const currentFileToDownload = downloadItems[downloadedFiles]
-          downloadedFiles += 1
+        while (currentFile < downloadItems.length) {
+          const currentFileToDownload = downloadItems[currentFile]
+          currentFile += 1
+
+          const startTime = performance.now()
           await this.downloadIndividualFile(
             index,
             currentFileToDownload.sourceLocation,
             currentFileToDownload.targetPath
           )
+
+          debug(
```

Comment:
You could wrap these debug calls in an if statement with `core.isDebug` so that we can skip the performance math when it's not needed. See https://github.com/actions/toolkit/tree/master/packages/core#logging